fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialogBottomSheet.kt

```diff
@@ -0,0 +1,234 @@
+                ),
+                textLabel = stringResource(R.string.translation_option_bottom_sheet_always_translate),
+            ),
+        )
+        add(
+            TranslationSwitchItem(
+                isEnabled = false,
+                onStateChange = {},
+                hasDivider = false,
+                contentDescription = getSwitchContentDescription(
+                    textSwitchId =
+                    R.string.translation_option_bottom_sheet_always_translate_in_language_content_description,
+                    additionalTextSwitchArgs = mutableListOf(
+                        stringResource(R.string.translation_option_bottom_sheet_spanish_language),
+                    ),
+                    isEnabled = false,
+                ),
+                textLabel = stringResource(
+                    id = R.string.translation_option_bottom_sheet_always_translate_in_language,
+                    formatArgs = arrayOf(stringResource(R.string.translation_option_bottom_sheet_spanish_language)),
+                ),
+            ),
+        )
+        add(
+            TranslationSwitchItem(
+                isEnabled = true,
+                onStateChange = {},
+                hasDivider = true,
+                contentDescription =
+                getSwitchContentDescription(
+                    textSwitchId =
+                    R.string.translation_option_bottom_sheet_never_translate_in_language_content_description,
+                    additionalTextSwitchArgs = mutableListOf(
+                        stringResource(R.string.translation_option_bottom_sheet_spanish_language),
+                    ),
+                    isEnabled = true,
+                ),
+                textLabel = stringResource(
+                    id = R.string.translation_option_bottom_sheet_never_translate_in_language,
+                    formatArgs = arrayOf(stringResource(R.string.translation_option_bottom_sheet_spanish_language)),
+                ),
+            ),
+        )
+        add(
+            TranslationSwitchItem(
+                isEnabled = true,
+                onStateChange = {},
+                hasDivider = true,
+                contentDescription =
+                getSwitchContentDescription(
+                    textSwitchId = R.string.translation_option_bottom_sheet_never_translate_site_content_description,
+                    isEnabled = true,
+                ),
+                textLabel = stringResource(R.string.translation_option_bottom_sheet_never_translate_site),
+            ),
+        )
+    }
+}
+
+/**
+ * Returns a content description for a switch list item based on the enabled value.
+ */
+@Composable
+fun getSwitchContentDescription(
+    @StringRes textSwitchId: Int,
+    additionalTextSwitchArgs: MutableList<String> = mutableListOf(),
+    isEnabled: Boolean,
+): String {
+    if (isEnabled) {
+        additionalTextSwitchArgs.add(stringResource(R.string.accessibility_on_switch))
```

Comment:
https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD?node-id=7221:187721&mode=design#572340901
The designer said "These are a bit tricky: the focus could look like it is, with both the label and the control focused together but announced as one element" . This is the reason that I put contentDescription. I can leave the default implementation it is way more easy for me , but design must know it .

Reply:
https://github.com/mozilla-mobile/firefox-android/assets/8118371/96449d5d-f259-4328-9771-e175002a8670

Reply:
This is how the switch is read with the default implementation without contentDescription . Two elements .

Reply:
Thanks for the video and the comment thread! In the video, the toggle element is missing a `name` or label (in the `name`, `role`, `value` [trifecta](https://www.afb.org/blog/entry/name-role-value)).

This can happen when the label and toggle are not programmatically linked or something is overriding the `name`. What is happening now is what happened to me with my earlier edited patch [here](https://github.com/mozilla-mobile/firefox-android/pull/4081#discussion_r1362376220), that advice now applies (but maybe not the reasoning).

I think that Figma thread is referring to the fact that they are okay if it is one focus-able element labeled (e.g., Focus on **row**: "Never translate, toggle on" or hiding the label from the focus order (e.g., Focus on **toggle**: "Never translate, toggle on"). We can figure out the grouping and focusable elements pretty quickly using the default implementation, lets get the base component and `name` situated first.

The inherited implementation is preferred because it will always use the TalkBack standard names and terminology for controls. (e.g., As a user, I have a visual expectation of how a toggle will present itself. As a user, I have an aural expectation of how a toggle will present itself. If we use non-standard content descriptions on standard components, we can break user expectations. Also worth mentioning, some screen reader users have the speech rate settings at 3x or much more the default speed, so a non-standard announcement sticks out even further.) 

When the label and element are programmatically linked, I expect they will show up as one element. (You can test this without TalkBack. If you tap a label and the component toggles/clicks, then the label and the component are programmatically tied. If they are programmatically tied and still have the wrong TalkBack announcement, then something is overriding the announcement. (e.g., something like `clearAndSetSemantics`, grouping incorrectly, or inheriting something strange.) If they are programmatically tied and showing as two elements, we can group the row as the focus-able element to ensure one element and will inherit the toggle semantic behavior.

Quick summary:
I suspect: 
1. `clearAndSetSemantics` is interfering with the `name` of the toggle 
2. OR else the label and toggle aren't connected programmatically
3. OR else something is overriding the label

We can solve focus grouping as a secondary problem, if fixing the `name` or label doesn't automatically connect the elements.

I'd suggest checking out how `Menu` -> `Accessibility` -> toggle for `Automatic font sizing` is setup too.